src/msg/Message.h

```diff
@@ -224,7 +224,16 @@ namespace bi = boost::intrusive;
 // XioMessenger diagnostic "ping pong" flag (resend msg when send completes)
 #define MSG_MAGIC_REDUPE       0x0100
 
-class Message : public RefCountedObject {
+class BasicParams{
+  public:
+    virtual const utime_t& get_recv_stamp() const =0;
+    virtual const utime_t& get_throttle_stamp() const =0;
+    virtual const utime_t& get_recv_complete_stamp() const =0;
+    virtual const utime_t& get_dispatch_stamp() const =0;
+    virtual ~BasicParams(){};
+};
```

Comment:
With the templated function I don't think we actually need a shared Params class like this — why did you create it?

Reply:
@gregsfortytwo I intended to use some template argument constraints to require that all params classes to be derived from BasicParams so that they would have to implement get_XXX_stamp() methods. However, I've just found that c++ concepts is still in progress as part of the Concepts Technical Specification that were included in the working draft of C++20, so I guess I should remove BasicParams class. Is this right? If it is, I will remove it.

Reply:
Yeah, I think so. We'll still fail to build if you try and use something that doesn't have those functions. :)